lib/led.js

```diff
@@ -23,27 +143,36 @@ function Led(opts) {
   var pinValue;
   var defaultLed;
   var state;
-  var isFirmata = true;
 
   if (!(this instanceof Led)) {
     return new Led(opts);
   }
 
+  var controller = null;
+  var err = null;
+
+  if (!opts || !opts.hasOwnProperty("controller")) {
+    controller = Controllers["Standard"];
+  }
+  else if (typeof opts.controller === "string") {
+    controller = Controllers[opts.controller];
+  }
+  else {
+    controller = opts.controller;
+  }
+
+  if (controller == null) {
+    controller = Controllers["Standard"];
+  }
+
+  Object.defineProperties(this, controller);
```

Comment:
None of this is necessary if you move the relevant parts of this: 

```js
  if (!opts || !opts.hasOwnProperty("controller")) {
    controller = Controllers["Standard"];
  }
  else if (typeof opts.controller === "string") {
    controller = Controllers[opts.controller];
  }
  else {
    controller = opts.controller;
  }

  if (controller == null) {
    controller = Controllers["Standard"];
  }

  Object.defineProperties(this, controller);
```

To some point after this: 

```js
  Board.Component.call(
    this, opts = Board.Options(opts)
  );
```
 

If you look in accelerometer.js, it's correct: 

```js
  var controller = null;
  var err = null;
  
  // ...
  
  Board.Component.call(
    this, opts = Board.Options(opts)
  );

  if (opts.controller && typeof opts.controller === "string") {
    controller = Controllers[opts.controller.toUpperCase()];
  } else {
    controller = opts.controller;
  }

  if (controller == null) {
    controller = Controllers["ANALOG"];
  }

  Object.defineProperties(this, controller);
```